internal/pkg/service/stream/sink/tablesink/storage/repository/hook.go

```diff
@@ -173,22 +174,44 @@ func (h *hook) NewUsedDiskSpaceProvider() UsedDiskSpaceProvider {
 	}
 }
 
-func (h *hook) DecorateFileStateTransition(atomicOp *op.AtomicOp[storage.File], _ time.Time, fileKey storage.FileKey, from, to storage.FileState) *op.AtomicOp[storage.File] {
+func (h *hook) DecorateFileStateTransition(atomicOp *op.AtomicOp[storage.File], fileKey storage.FileKey, from, to storage.FileState) *op.AtomicOp[storage.File] {
 	// Move statistics to the target storage level, if needed
 	fromLevel := from.Level()
 	toLevel := to.Level()
 	if fromLevel != toLevel {
-		atomicOp.AddFrom(h.stats.MoveAll(fileKey, fromLevel, toLevel))
+		atomicOp.AddFrom(h.stats.MoveAll(fileKey, fromLevel, toLevel, func(value *statistics.Value) {
+			// There is actually no additional compression, when uploading slice to the staging storage
+			if toLevel == storage.LevelStaging {
+				value.StagingSize = value.CompressedSize
+			}
```

Comment:
There is `StagingSize` field, but it was always empty.

In the actual implementation, there is no additional compression in the Staging storage.

The slices are compressed locally (if enabled), and directly uploaded to the S3/ABS/GCS staging storage.

The field exists for the forward compatibility.